stm32_base/hal/timer_stm32/timer_device_stm32.c

```diff
@@ -309,15 +327,16 @@ uint32_t timer_device_get_count(size_t channel)
         return 0;
 }
 
-uint32_t ticks_to_us(uint16_t ticks)
+static uint32_t ticks_to_us(const size_t chan, const uint16_t ticks)
 {
-        /* 1 tick == 2us */
-        return (uint32_t) ticks << 1;
+        /* Our pre-scalars are setup to never have less than 1 tick per us */
+        const uint32_t us_per_tick = get_clk_speed(chan) / 1000000;
+        return us_per_tick ? g_config[chan].prescaler * ticks / us_per_tick : 0;
```

Comment:
I was leaving that to the higher level methods.  The way I see it, methods designed to be invoked from other c files should do the validation.  Otherwise it becomes redundant.